lib/resolve-imports-exports.js

```diff
@@ -0,0 +1,208 @@
+var path = require('path');
+var startsWith = require('string.prototype.startswith');
+
+function validateExports(exports, basePath) {
+    var isConditional = true;
+
+    if (typeof exports === 'object' && !Array.isArray(exports)) {
+        var exportKeys = Object.keys(exports);
+
+        for (var i = 0; i < exportKeys.length; i++) {
+            var isKeyConditional = exportKeys[i][0] !== '.';
+            if (i === 0) {
+                isConditional = isKeyConditional;
+            } else if (isKeyConditional !== isConditional) {
+                var err = new Error('Invalid package config ' + basePath + path.sep + 'package.json, '
+                    + '"exports" cannot contain some keys starting with \'.\' and some not. '
+                    + 'The exports object must either be an object of package subpath keys '
+                    + 'or an object of main entry condition name keys only.');
+                err.code = 'ERR_INVALID_PACKAGE_CONFIG';
+                throw err;
+            }
+        }
+    }
+
+    if (isConditional) {
+        return { '.': exports };
+    } else {
+        return exports;
+    }
+}
+
+function validateConditions(names) {
+    // If exports contains any index property keys, as defined in ECMA-262 6.1.7 Array Index, throw an Invalid Package Configuration error.
+
+    for (var i = 0; i < names.length; i++) {
+        var name = names[i];
+        var nameNum = Number(name);
+
+        if (String(nameNum) === name && nameNum >= 0 && nameNum < 0xFFFFFFFF) {
+            var err = new Error('"exports" cannot contain numeric property keys');
```

Comment:
It'd be nice if we could replicate node's error message here:
```suggestion
            var err = new Error('Invalid package config ' + pathToPackageJSON + '. "exports" cannot contain numeric property keys.');
```